server/channels/app/post_test.go

```diff
@@ -2068,6 +2069,63 @@ func TestCountMentionsFromPost(t *testing.T) {
 		assert.Equal(t, 1, count)
 	})
 
+	t.Run("should not include comments made before the given post when rootPost is inaccessible", func(t *testing.T) {
+		th := Setup(t).InitBasic()
+		defer th.TearDown()
+
+		th.App.Srv().SetLicense(model.NewTestLicense("cloud"))
+
+		user1 := th.BasicUser
+		user2 := th.BasicUser2
+
+		channel := th.CreateChannel(th.Context, th.BasicTeam)
+		th.AddUserToChannel(user2, channel)
+
+		user2.NotifyProps[model.CommentsNotifyProp] = model.CommentsNotifyAny
+
+		post1, err := th.App.CreatePost(th.Context, &model.Post{
+			UserId:    user1.Id,
+			ChannelId: channel.Id,
+			Message:   "test1",
+		}, channel, false, true)
+		require.Nil(t, err)
+		_, err = th.App.CreatePost(th.Context, &model.Post{
+			UserId:    user2.Id,
+			ChannelId: channel.Id,
+			RootId:    post1.Id,
+			Message:   "test2",
+		}, channel, false, true)
+		require.Nil(t, err)
+		post3, err := th.App.CreatePost(th.Context, &model.Post{
+			UserId:    user1.Id,
+			ChannelId: channel.Id,
+			Message:   "test3",
+		}, channel, false, true)
+		require.Nil(t, err)
+		_, err = th.App.CreatePost(th.Context, &model.Post{
```

Comment:
Do we want to perhaps sleep for couple of ms to prevent post3.CreateAt to be the same as the next post? It's possible to have this in a busy CI system.